src/libraries/System.Private.DataContractSerialization/src/System/Runtime/Serialization/Json/JsonReaderDelegator.cs

```diff
@@ -153,37 +153,29 @@ internal static DateTime ParseJsonDateInDefaultFormat(string originalDateTimeVal
                 throw new FormatException(SR.Format(SR.JsonInvalidDateTimeString, originalDateTimeValue, JsonGlobals.DateTimeStartGuardWriter, JsonGlobals.DateTimeEndGuardWriter));
             }
 
-            string ticksvalue = dateTimeValue.Substring(6, dateTimeValue.Length - 8);
+            ReadOnlySpan<char> ticksvalue = dateTimeValue.AsSpan(6, dateTimeValue.Length - 8);
             long millisecondsSinceUnixEpoch;
             DateTimeKind dateTimeKind = DateTimeKind.Utc;
-            int indexOfTimeZoneOffset = ticksvalue.IndexOf('+', 1);
+            int indexOfTimeZoneOffset = ticksvalue.Slice(1).IndexOf('+');
```

Comment:
Can this instead use `ticksValue.Slice(1).IndexOfAny('+', '-')`?

Reply:
I originally wrote that, and then I undid it as I was concerned I might be changing behavior, since in theory it could end up finding a '-' before a '+'.